flink-core/src/test/java/org/apache/flink/configuration/ConfigOptionTest.java

```diff
@@ -86,4 +87,32 @@ public void testDeprecationFlagForMixedAlternativeKeys() {
 		assertThat(fallbackKeys, containsInAnyOrder("fallback1", "fallback2"));
 		assertThat(deprecatedKeys, containsInAnyOrder("deprecated1", "deprecated2"));
 	}
+
+	@Test
+	public void testDeprecationForDeprecatedKeys() {
+		String[] deprecatedKeys = new String[] { "deprecated1", "deprecated2" };
+		final ConfigOption<Integer> optionWithDeprecatedKeys = ConfigOptions
+			.key("key")
+			.defaultValue(0)
+			.withDeprecatedKeys(deprecatedKeys)
+			.withFallbackKeys("fallback1");
+
+		assertTrue(optionWithDeprecatedKeys.hasDeprecatedKeys());
+		int counter = 0;
+		for (final String deprecatedKey : optionWithDeprecatedKeys.deprecatedKeys()) {
+			counter++;
+			assertTrue(Arrays.stream(deprecatedKeys).anyMatch(item -> item.equals(deprecatedKey)));
+		}
+
+		assertEquals(2, counter);
+
+		final ConfigOption<Integer> optionWithFallbackKeys = ConfigOptions
+			.key("key")
+			.defaultValue(0)
+			.withFallbackKeys("fallback1");
```

Comment:
Let's split this test up into a separate one. It is usually better to have smaller and more targeted tests.